eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -95,6 +113,103 @@ function GetPackageLookup($packageList) {
   return $packageLookup
 }
 
+function generate-service-level-readme($readmeBaseName, $pathPrefix, $clientPackageInfo, $mgmtPackageInfo, $serviceName) {
+  # Add ability to override
+  # Fetch the service readme name
+  $monikers = @("latest", "preview")
+
+  for($i=0; $i -lt $monikers.Length; $i++) {
+    $serviceReadme = "$DocRepoLocation/$pathPrefix/$($monikers[$i])/$readmeBaseName.md"
+    $overrideReadme = "$DocRepoLocation/$pathPrefix/$($monikers[$i])/$readmeBaseName-override.md"
+    if (Test-Path $overrideReadme) {
+      Copy-Item $overrideReadme -Destination $serviceReadme
+      return
+    }
+    if(Test-Path $serviceReadme) {
+      continue
+    }
+    update-service-readme -readmePath $serviceReadme -moniker $monikers[$i] -clientPackageInfo $clientPackageInfo -mgmtPackageInfo $mgmtPackageInfo -serviceName $serviceName
+  } 
+}
+
+function update-service-readme($readmePath, $moniker, $clientPackageInfo, $mgmtPackageInfo, $serviceName)
+{
+  # Add metadata header
+  $lang = &$GetLanguageDisplayNameFn
+  $langTitle = "Azure $serviceName SDK for $lang"
+  $langDescription = "Reference for Azure $serviceName SDK for $lang"
+  $githubUrl = &$GetLanguageGithubUrlFn
+  # Github url for source code: e.g. https://github.com/Azure/azure-sdk-for-js
+  $serviceBaseName = $serviceName.ToLower().Replace(' ', '').Replace('/', '-')
+  $author = GetPrimaryCodeOwner -TargetDirectory "/sdk/$serviceBaseName/"
+  if (!$author) {
+    LogError "Cannot fetch the author from CODEOWNER file."
+    $author = "sima-zhu"
+    $msauthor = "sizhu"
+  }
+  elseif ($TenantId -and $ClientId -and $ClientSecret) {
+    $msauthor = GetMsAliasFromGithub -TenantId $TenantId -ClientId $ClientId -ClientSecret $ClientSecret -GithubUser $author
+  }
+  # Default value
+  if (!$msauthor) {
+    LogWarning "No ms.author found for $author. Please check your app credential."
+    $msauthor = $author
+  }
+  $date = Get-Date -Format "MM/dd/yyyy"
+  if ($clientPackageInfo -and $clientPackageInfo.MSDocService) {
+    # Use MSDocService in csv metadata to override the service directory
+    # TODO: Use taxonomy for service name -- https://github.com/Azure/azure-sdk-tools/issues/1442
+    $service = $clientPackageInfo.MSDocService
+  }
+  elseif ($mgmtPackageInfo -and $mgmtPackageInfo.MSDocService) {
+    $service = $mgmtPackageInfo.MSDocService
+  }
+  Write-Host "The service of package: $service"
+  $header = @"
+---
+title: $langTitle
+description: $langDescription
+author: $author
+ms.author: $msauthor
+ms.date: $date
+ms.topic: reference
+ms.devlang: $lang
+ms.service: $service
+---
+"@
+
+  Add-Content -Path $readmePath -Value $header
+
+  # Add tables, seperate client and mgmt.
+  $tableHeader = "## $langTitle - $moniker"
```

Comment:
Do you have example of what this table will look like?

Reply:
Update in description